File: Java/HashHelper.java
import javax.crypto.Mac;
import javax.crypto.spec.SecretKeySpec;
import java.security.InvalidKeyException;
import java.security.NoSuchAlgorithmException;

class HashHelper {
    public static byte[] ComputeSHA256HMAC(final byte[] data, final byte[] key) {
        return HashHelper.ComputeSHA256HMAC(data, key, false);
    }

    public static byte[] ComputeSHA256HMAC(final byte[] data, final byte[] key, boolean silent) {
        if (!silent) {
            System.out.println("Computation of HMAC-SHA-256 hash with " + data.length + " bytes started");
        }

        if (data.length <= 0 || key.length <= 0) {
            throw new RuntimeException("No data from which to calculate hmac");
        }

        byte[] finalData;
        try {
            final String hmacType = "HmacSHA256";
            Mac sha256HMAC = Mac.getInstance(hmacType);
            SecretKeySpec keySpec = new SecretKeySpec(key, hmacType);
            sha256HMAC.init(keySpec);
            finalData = sha256HMAC.doFinal(data);
        } catch (NoSuchAlgorithmException | InvalidKeyException e) {
            throw new RuntimeException("Computation of HMAC-SHA-256 failed", e);
        }

        if (!silent) {
            System.out.println("HMAC-SHA-256 computation finished");
        }

        return finalData;
    }
}

File: C++/HashHelper.cpp
#include <iostream>
#include <stdexcept>
#include "HashHelper.h"
#include "sha.h"
#include "hmac.h"
#include "filters.h"

bool HashHelper::ComputeSHA256HMAC(const std::vector<byte>& data, const std::vector<byte>& key, std::vector<byte>& hmac, bool silent)
{
	if (!silent)
	{
		std::cout << "Computation of HMAC-SHA-256 hash with " << data.size() << " bytes started" << std::endl;
	}

	// check if there is data to hash
	if (data.size() > 0 && key.size() > 0)
	{
		hmac.clear();
		hmac.resize(CryptoPP::SHA256::DIGESTSIZE);

		try
		{
			// construct HMAC object used to create hash and ...
			const byte *plainKey = key.data();
			CryptoPP::HMAC<CryptoPP::SHA256> hmacForFilter(plainKey, key.size());

			// ... use it to transform the input
			const byte *in = data.data();
			byte *out = hmac.data();
			CryptoPP::ArraySource(in, data.size(), true, new CryptoPP::HashFilter(hmacForFilter, new CryptoPP::ArraySink(out, CryptoPP::SHA256::DIGESTSIZE)));
		}
		catch (const std::exception&)
		{
			std::cerr << "Computation of HMAC-SHA-256 failed" << std::endl;
			throw;
		}

		if (!silent)
		{
			std::cout << "HMAC-SHA-256 computation finished" << std::endl;
		}
		return true;
	}
	else
	{
		throw std::runtime_error("No data from which to calculate hmac");
	}
}

bool HashHelper::ComputeSHA512HMAC(const std::vector<byte>& data, const std::vector<byte>& key, std::vector<byte>& hmac, bool silent)
{
	if (!silent)
	{
		std::cout << "Computation of HMAC-SHA-512 hash with " << data.size() << " bytes started" << std::endl;
	}

	// check if there is data to hash 
	if (data.size() > 0 && key.size() > 0)
	{
		hmac.clear();
		hmac.resize(CryptoPP::SHA512::DIGESTSIZE);

		try
		{
			// construct HMAC object used to create hash and ...
			const byte *plainKey = key.data();
			CryptoPP::HMAC<CryptoPP::SHA512> hmacForFilter(plainKey, key.size());

			// ... use it to transform the input
			const byte *in = data.data();
			byte *out = hmac.data();
			CryptoPP::ArraySource(in, data.size(), true, new CryptoPP::HashFilter(hmacForFilter, new CryptoPP::ArraySink(out, CryptoPP::SHA512::DIGESTSIZE)));
		}
		catch (const std::exception&)
		{
			std::cerr << "Computation of HMAC-SHA-512 failed" << std::endl;
			throw;
		}

		if (!silent)
		{
			std::cout << "HMAC-SHA-512 computation finished" << std::endl;
		}
		return true;
	}
	else
	{
		throw std::runtime_error("No data from which to calculate hmac");;
	}
}